test/e2e/network/network_policy.go

```diff
@@ -277,6 +277,39 @@ var _ = SIGDescribe("NetworkPolicy", func() {
 			testCanConnect(f, f.Namespace, "client-a", service, 80)
 			testCanConnect(f, f.Namespace, "client-b", service, 81)
 		})
+
+		It("should allow ingress access on one named port [Feature:NetworkPolicy]", func() {
+			policy := &networking.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "allow-client-a-via-named-port-ingress-rule",
+				},
+				Spec: networking.NetworkPolicySpec{
+					// Apply this policy to the Server
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-name": podServer.Name,
+						},
+					},
+					// Allow traffic to only one named port: "serve-80".
+					Ingress: []networking.NetworkPolicyIngressRule{{
+						Ports: []networking.NetworkPolicyPort{{
+							Port: &intstr.IntOrString{Type: intstr.String, StrVal: "serve-80"},
+						}},
+					}},
+				},
+			}
+
+			policy, err := f.InternalClientset.Networking().NetworkPolicies(f.Namespace.Name).Create(policy)
```

Comment:
Why use the InternalClientset and not the versioned one?
I see that this test uses both at various places.

Using versioned clientset is better since it immunes you from internal API changes

Reply:
Yea, would be great to switch to the versioned client. That said, the rest of the tests use the internal clientset AFAICT and I don't want to burden @bcreane's PR with thus just because he touched the code :)

I think we can address that in its own PR. WDYT?

Reply:
it's a good suggestion to move to the versioned client set. I can get another PR out today if that approach is acceptable.

Reply:
Thanks! Updating all references in another PR sounds good.